python/hawkey/query-py.cpp

```diff
@@ -694,6 +694,36 @@ get_advisory_pkgs(_QueryObject *self, PyObject *args)
     return advisoryPkgVectorToPylist(advisoryPkgs);
 }
 
+static PyObject *
+filter_userinstalled(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    HyQuery self_query_copy = new libdnf::Query(*((_QueryObject *) self)->query);
+    const char *kwlist[] = {"swdb", NULL};
+    PyObject *pySwdb;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O", (char **)kwlist, &pySwdb)) {
+        return NULL;
```

Comment:
Memory leak. `self_query_copy` is allocated on heap.

Reply:
Fixed